python/src/scipp/neutron/instrument_view.py

```diff
@@ -372,14 +409,131 @@ def __init__(self,
+            self.nfaces * 3,
+            axis=0).reshape(self.nfaces * self.ndets, 3)
+
+        vertexcolors = np.zeros([self.ndets * self.nverts, 3],
+                                dtype=np.float32)
+
+        mesh_geometry = self.p3.BufferGeometry(attributes=dict(
+            position=self.p3.BufferAttribute(np.zeros(
+                [self.nverts * self.ndets, 3], dtype=np.float32),
+                                             normalized=False),
+            index=self.p3.BufferAttribute(faces.ravel(), normalized=False),
+            color=self.p3.BufferAttribute(vertexcolors),
+        ))
+
+        mesh_material = self.p3.MeshBasicMaterial(vertexColors='VertexColors',
+                                                  transparent=True)
+
+        mesh = self.p3.Mesh(geometry=mesh_geometry, material=mesh_material)
+
+        return mesh_geometry, mesh_material, mesh
+
+    def create_points_geometry(self):
+        """
+        Make a simple PointsGeometry for the 'Fast' rendering mode
+        """
+        points_geometry = self.p3.BufferGeometry(
+            attributes={
+                'position':
+                self.p3.BufferAttribute(array=self.det_pos.values),
+                'color':
+                self.p3.BufferAttribute(array=np.zeros(
+                    [self.det_pos.shape[0], 3], dtype=np.float32))
+            })
+        points_material = self.p3.PointsMaterial(vertexColors='VertexColors',
+                                                 size=self.camera_pos * 0.05,
+                                                 transparent=True)
+        points = self.p3.Points(geometry=points_geometry,
+                                material=points_material)
+        return points_geometry, points_material, points
+
+    def get_detector_vertices_and_faces(self):
+        if "instrument_name" in self.data_arrays[self.key].attrs:
+            instrument_name = self.data_arrays[
+                self.key].attrs["instrument_name"].value
+        else:
+            instrument_name = ""
+        cylindrical_major_axis = {"loki": "x"}
+        instrument_name = instrument_name.lower()
+        axis = None
+        for name in cylindrical_major_axis:
+            if instrument_name.startswith(name):
+                axis = cylindrical_major_axis[name]
+                break
+        if axis is not None:
+            sq2 = 0.25 * np.sqrt(2.0)
+            vertices = np.array(
+                [[0.0, 0.0, -0.5], [0.5, 0.0, -0.5], [sq2, sq2, -0.5],
+                 [0.0, 0.5, -0.5], [-sq2, sq2, -0.5], [-0.5, 0.0, -0.5],
+                 [-sq2, -sq2, -0.5], [0.0, -0.5, -0.5], [sq2, -sq2, -0.5],
+                 [0.0, 0.0, 0.5], [0.5, 0.0, 0.5], [sq2, sq2, 0.5],
+                 [0.0, 0.5, 0.5], [-sq2, sq2, 0.5], [-0.5, 0.0, 0.5],
+                 [-sq2, -sq2, 0.5], [0.0, -0.5, 0.5], [sq2, -sq2, 0.5]],
+                dtype=np.float32)
+
+            # By default, we assume major axis of cylinder is z. If it is not
```

Comment:
How do you know pixels are cylinders? Just based on instrument whitelist?

Reply:
Yes. Any instrument which is in this `dict` will be cylinders, otherwise it's boxes
https://github.com/scipp/scipp/pull/1049/files#diff-a37c4db6df31e257d37ece7462280c79R483